components/log4j/org.wso2.carbon.analytics.shared.data.agents.log4j/src/main/java/org/wso2/carbon/analytics/shared/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -430,8 +433,18 @@ private void publishLogEvent(TenantDomainAwareLoggingEvent event) throws ParseEx
             if (event.getThrowableInformation() != null) {
                 arbitraryDataMap.put(columns[8], stacktrace);
             }
-            Event laEvent = new Event(streamDef, date.getTime(), null, null, new String[]{tenantDomain}, arbitraryDataMap);
-            dataPublisher.publish(laEvent);
+            Event logEvent = new Event(streamDef, date.getTime(), null, null, new String[]{tenantDomain},
+                    arbitraryDataMap);
+            if (dataPublisher != null) {
+                dataPublisher.publish(logEvent);
+            } else {
+                publisherInitializer();
```

Comment:
The method names should be something with verbs, and hence please rename the method as initializePublisher() instead of publisherInitializer which is a noun.